File: math/dm1/dm1.py
#!/usr/bin/python3

n = 0
ls = [ ]
for n in range(0, 100):
    ls.append(n + 1)

l = 0
for l in range(1, 100):
    for n in range(0, 100-l):
        ls[n] = ls[n] + ls[n+1]
print("Result : ", ls[0])
print("Theo   : ", 2**98 * 101)

File: math/dm1/dm1.pl
#!/usr/bin/perl

use strict;
use warnings;

my @line;
# Init : first line.
for(my $i = 0; $i < 100; ++$i) {
    $line[$i] = $i + 1;
}

# For each line.
for(my $l = 1; $l < 100; ++$l) {
    my $n = 100 - $l;
    for(my $i = 0; $i < $n; ++$i) {
        print "$line[$i] ";
        $line[$i] = $line[$i] + $line[$i + 1];
    }
    print "$line[$n] \n";
}

print "$line[0]\n\n";

my $n = 2**98 * 101;
print "Theo : $n\n";
print "Exp  : $line[0]\n";
print "Verified.\n" if($line[0] == $n);

